drivers/sensor/veml7700/veml7700.c

```diff
@@ -535,30 +499,80 @@ static int veml7700_init(const struct device *dev)
 	return 0;
 }
 
+static int veml7700_init(const struct device *dev)
+{
+	/* If the device is behind a power domain, it will start in
+	 * PM_DEVICE_STATE_OFF.
+	 */
+	if (pm_device_on_power_domain(dev)) {
```

Comment:
`pm_device_on_power_domain` says nothing about whether the device is powered or not, so cannot be used to determine whether `pm_device_init_off` is correct or not.

Look into `pm_device_driver_init`, which always puts the device into the correct state for the hardware configuration.